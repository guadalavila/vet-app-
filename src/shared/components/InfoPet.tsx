import React from 'react';
import { StyleSheet, View } from 'react-native';
import { GlobalStyles } from '../utils/styles';
import CustomText from './CustomText';
import { size } from '../utils/size';
import { typography } from '../utils/typography';

interface IInfoPetProps {
    label: string;
    info: string;
}
const InfoPet = ({ label, info }: IInfoPetProps) => {
    return (
        <View style={[GlobalStyles.rowBetween, styles.container]}>
            <CustomText style={styles.label}>{label}:</CustomText>
            <CustomText>{info}</CustomText>
        </View>
    );
};

export default InfoPet;

const styles = StyleSheet.create({
    container: {
        marginHorizontal: size.XXL,
        marginVertical: size.L,
    },
    label: {
        fontWeight: 'bold',
        fontSize: typography.size.S,
    },
});
