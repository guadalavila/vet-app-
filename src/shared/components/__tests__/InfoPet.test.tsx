import { render, screen } from '@testing-library/react-native';
import React from 'react';
import InfoPet from '../InfoPet';

describe('InfoPet', () => {
    test('should match snapshot', () => {
        const container = render(<InfoPet label='Nombre' info='Boby' />);
        expect(container).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<InfoPet label='Nombre' info='Boby' />);
        expect(screen.getByTestId('InfoPet')).toBeTruthy();
    });

    test('should render texts', () => {
        render(<InfoPet label='Nombre' info='Boby' />);
        expect(screen.getByText('Nombre:')).toBeTruthy();
        expect(screen.getByText('Boby')).toBeTruthy();
    });
});
