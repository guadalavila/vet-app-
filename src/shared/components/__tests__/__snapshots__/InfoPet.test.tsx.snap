// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPet should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "marginHorizontal": 14,
        "marginVertical": 8,
      },
    ]
  }
  testID="InfoPet"
>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Nombre
    :
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Object {},
      ]
    }
  >
    Boby
  </Text>
</View>
`;
